pytest_tank_test/tk_fixtures.py

```diff
@@ -0,0 +1,232 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pytest
+import os
+from tk_toolchain.authentication import get_toolkit_user
+from tk_toolchain.testing import create_unique_name
+
+
+@pytest.fixture(scope="session")
+def tk_test_shotgun():
+    """
+    Getting credentials from TK_TOOLCHAIN
+    """
+    sg = get_toolkit_user().create_sg_connection()
+
+    return sg
+
+
+@pytest.fixture(scope="session")
+def tk_test_current_user(tk_test_shotgun):
+    """
+    Get current user
+
+    :returns: The current user id and name
+    """
+    user = get_toolkit_user()
+    username = tk_test_shotgun.find_one(
+        "HumanUser", [["login", "is", str(user)]], ["name"]
+    )
+
+    return username
+
+
+@pytest.fixture(scope="session")
+def tk_test_create_project(tk_test_shotgun):
+    """
+    Generates a fresh Shotgun Project to use with the UI Automation.
+
+    :returns: Current project name and id
+    """
+    # Create or update the integration_tests local storage with the current test run
+    storage_name = "Toolkit UI Automation"
```

Comment:
Ideally, we would create a unique name here, so if two tests run on different repos at the same time, each will have their unique storage location.

Reply:
I know that in practice Azure pipeline repos are always cloned at the same location, but it could change in the future and there's no real downside to creating a unique name here, so we shouldn't hesitate to do it. It might save us some headaches down the road.

Reply:
I can't use a unique name for the store because Workfiles2 automation is using an advanced custom config and I need to hardcode the storage name in roots.yml

But I'll add a comment inside the code about it